File: Lista11_Python/Ex03_PegadaDeCarbono/escola.py
from predio import *

class Escola(Predio):

    def __init__(self, nFrequentadores, mediaLixoPorFrequentador, consumoGasDiario, nSalas):
        super().__init__(nFrequentadores, mediaLixoPorFrequentador, consumoGasDiario)
        self.nSalas = nSalas

    def __str__(self):
        strSalas = str(self.nSalas)

        string = super().__str__()
        string = string+"\nnSalas: "+strSalas+"\n"

        return string
    
    def getNSalas(self):
        return self.nSalas

    def getPegadaCarbono(self):
        pCarbono = super().getPegadaCarbono()
        pCarbono *= 0.2 * self.nSalas
        return pCarbono

File: Extra/Ruby/PegadaDeCarbono/escola.rb
require './predio.rb'

class Escola < Predio

  attr_accessor :n_salas

  def initialize(n_frequentadores, media_lixo_por_morador, consumo_gas_diario, n_salas)
    @n_frequentadores = n_frequentadores
    @media_lixo_por_morador = media_lixo_por_morador
    @consumo_gas_diario = consumo_gas_diario
    @n_salas = n_salas
  end

  def print
    puts "ESCOLA"
    super()
    puts "n_salas: #{@n_salas}"
    puts "--------------------"
  end

  def getPegadaCarbono
    pCarbono = (@media_lixo_por_morador * @n_frequentadores * 0.05) + @consumo_gas_diario * 0.1 + @n_salas*0.30 
    return pCarbono
  end

end
